### PR TITLE
Fix duplication of codeblock elements

### PR DESCRIPTION
**Pull request by SimonBrandner**
Fixes [#16421](https://github.com/vector-im/element-web/issues/16421)

### CODE REVIEW

File: src/components/views/messages/TextualBody.js [hunk 1]
```diff
@@ -96,6 +96,9 @@ export default class TextualBody extends React.Component {
             const pres = ReactDOM.findDOMNode(this).getElementsByTagName("pre");
             if (pres.length > 0) {
                 for (let i = 0; i < pres.length; i++) {
+                    // If there already is a div wrapping the codeblock we want to skip this.
+                    // This happens after the codeblock was edited.
+                    if (pres[i].parentNode.className == "mx_EventTile_pre_container") continue;
```
Comment (jryans) [inline src/components/views/messages/TextualBody.js:101]: Thanks, this covers the `pre` portion, but in testing this fix, it looks like the `code` part in the next block also duplicates as well. Perhaps we can test `codes[i]` inside the `setTimeout` for the presence of the `hljs` class, and skip if it exists?

While we're there, the `for (let i = 0; i < pres.length; i++) { ... }` loop inside the `setTimeout` seem incorrect and redundant given the other loop outside the `setTimeout`, so I believe it can be removed.

**Review by jryans — CHANGES_REQUESTED** [review]
Thanks for taking this on! 😄

File: src/components/views/messages/TextualBody.js (hunk 1 above)
Comment (SimonBrandner) [reply src/components/views/messages/TextualBody.js:101]: I am sorry for the weird nested loops... I've fixed that now. 

I don't think I see where the `code` duplication could happen and I can't reproduce it... Maybe I am missing something?

File: src/components/views/messages/TextualBody.js [hunk 2]
```diff
@@ -110,16 +113,12 @@ export default class TextualBody extends React.Component {
             // Highlight code
             const codes = ReactDOM.findDOMNode(this).getElementsByTagName("code");
             if (codes.length > 0) {
-                for (let i = 0; i < codes.length; i++) {
-                    // Do this asynchronously: parsing code takes time and we don't
```
Comment (jryans) [inline src/components/views/messages/TextualBody.js:114]: Let's keep this comment, it seems like a helpful reminder to future readers on why this `setTimeout` is here.

File: src/components/views/messages/TextualBody.js (hunk 1 above)
Comment (jryans) [reply src/components/views/messages/TextualBody.js:101]: Just like with the `pre`s, the `this._highlightCode` portion below is re-run on each edit. Here's a screenshot of what I see when editing a code block with this PR. Maybe you need to test with a code block that forces syntax highlighting?

For example, send a code block starting with `` ```js `` (instead of just `` ``` ``), which forces JavaScript syntax highlighting. (There is a default off option to detect the language for all code blocks, but as mentioned, it's off by default, so probably is not running for you.)

If I edit such a block, all of the `hljs-*` tags to style the code are duplicated:

image

**Review by jryans — CHANGES_REQUESTED** [review]

File: src/components/views/messages/TextualBody.js [hunk 3]
```diff
@@ -113,6 +113,7 @@ export default class TextualBody extends React.Component {
             // Highlight code
             const codes = ReactDOM.findDOMNode(this).getElementsByTagName("code");
             if (codes.length > 0) {
+                // Do this asynchronously: parsing code takes time and we don't
```
Comment (jryans) [inline src/components/views/messages/TextualBody.js:116]: Don't forget the second comment line as well. 😉

**Review by jryans — APPROVED** [review]
Thanks for taking this on, looks great to me! 😄
